lib/refocus/samples/collector.rb

```diff
@@ -19,11 +20,16 @@ def add(name:, aspect:, value:nil, message_body:nil, message_code:nil, related_l
 
       def upsert_bulk
         result = json(http.post("upsert/bulk", body: samples, expects: 200))
+        @job_id = result["jobId"]
```

Comment:
I don't like the pattern of storing state in the Collector class, it limits the ways in which it can be used. This pattern prevents us from using the same Collector to submit more than one bulk requests at a time if we want to check the statuses.

I'd prefer to parameterize the check_status method with a `job_id` parameter. This way, Collector can remain stateless.

eg:

```ruby
job_id_1 = collector.upsert_bulk["jobId"]
// ... do some more stuff here
job_id_2 = collector.upsert_bulk["jobId"]

status_1 = collector.check_status(job_id_1)
status_2 = collector.check_status(job_id_2)
```